front_end/src/components/history/ChangeReservation.js

```diff
@@ -0,0 +1,302 @@
+                      Select Dates
+                    </Button>
+                  )}
+                </DialogActions>
+                {this.state.showChange === true &&
+                (expansionData.total / expansionData.numberOfNights) *
+                  this.state.days -
+                  expansionData.total >
+                  0 ? (
+                  <div>
+                    <CardContent>
+                      <h4 style={{ marginTop: "1%" }}>
+                        Payment for Extra{" "}
+                        {this.state.days - expansionData.numberOfNights}{" "}
+                        Night(s) : $
+                        {(
+                          (expansionData.total / expansionData.numberOfNights) *
+                            this.state.days -
+                          expansionData.total
+                        ).toFixed(2)}
+                      </h4>
+                      <hr />
+                      <TextFieldGroup
+                        placeholder="Name on Card"
+                        name="nameOnCard"
+                      />
+
+                      <CardElement
+                        name="paymentField"
+                        onChange={this.stripeValidate}
+                        id="sample-input"
+                      />
+                    </CardContent>
+                  </div>
+                ) : null}
+                {this.state.showChange === true &&
+                (expansionData.total / expansionData.numberOfNights) *
+                  this.state.days -
+                  expansionData.total <
+                  0 ? (
+                  <DialogTitle>
+                    <br /> <br />
+                    You will be automatically refunded $
+                    {(-(
+                      (expansionData.total / expansionData.numberOfNights) *
+                        this.state.days -
+                      expansionData.total
+                    )).toFixed(2)}
+                    .
+                  </DialogTitle>
+                ) : null}
+              </Grid>
+            </Grid>
+            <DialogActions>
+              {this.state.newCheckIn == null ||
+              this.state.newCheckOut == null ||
+              this.state.showChange == false ? null : (
+                <Button
+                  onClick={() => this.onChangeClick(id, expansionData)}
+                  variant="outlined"
+                  color="primary"
+                  className={classes.button}
+                >
+                  Confirm Change
+                </Button>
+              )}
+
+              <Button
+                onClick={this.handleChangeClose}
+                variant="outlined"
+                color="secondary"
+                className={classes.button}
+              >
+                Cancel Change
+              </Button>
+            </DialogActions>
+          </Grid>
+        </Dialog>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({});
```

Comment:
Please leave a 'TODO' comment here if this variable needs to be changed in future stories.

Reply:
It will not need to be changed in future stories.